pygrametl/__init__.py

```diff
@@ -444,23 +444,38 @@ def now(ignoredtargetconn=None, ignoredrow=None, ignorednamemapping=None):
 
 
 def ymdparser(ymdstr):
-    """Convert a string of the form 'yyyy-MM-dd' to a datetime.date.
+    """Convert input of the form 'yyyy-MM-dd' to a datetime.date.
 
        If the input is None, the return value is also None.
+       If the input is a datetime.date, it is returned.
+       If the input is a datetime.datetime, its date is returned.
+       Else the input is cast to str and quotes are stripped before it is
+       split into the different parts needed to create a datetime.date.
     """
     if ymdstr is None:
         return None
+    elif isinstance(ymdstr, date):
+        return ymdstr
+    elif isinstance(ymdstr, datetime):
+        return ymdstr.date()
+    ymdstr = str(ymdstr).strip("'\"")
```

Comment:
It is unclear to me why `strip()` is added as it does not seem to be present in the previous version of `ymdparser()` and `ymdhmsparser()`.

Reply:
Connections may use their own date(time) formats which can include quotes when cast to a str. Thus, this was included (similarly to how strip is used on l. [1161](https://github.com/chrthomsen/pygrametl/blob/5d03418b5c8f6d86f0d4eac7917ca1889863839a/pygrametl/tables.py#L1161) and [1162](https://github.com/chrthomsen/pygrametl/blob/5d03418b5c8f6d86f0d4eac7917ca1889863839a/pygrametl/tables.py#L1162) in tables.py)